index.js

```diff
@@ -28,6 +27,130 @@ var warn = function warn(message) {
+  },
+  czech: function (n) {
+    if (n === 1) { return 0; }
+    return (n >= 2 && n <= 4) ? 1 : 2;
+  },
+  polish: function (n) {
+    if (n === 1) { return 0; }
+    return n % 10 >= 2 && n % 10 <= 4 && (n % 100 < 10 || n % 100 >= 20) ? 1 : 2;
+  },
+  icelandic: function (n) { return (n % 10 !== 1 || n % 100 === 11) ? 1 : 0; }
+};
+
+// Mapping from pluralization group to individual locales.
+var pluralTypeToLanguages = {
+  arabic: ['ar'],
+  chinese: ['fa', 'id', 'ja', 'ko', 'lo', 'ms', 'th', 'tr', 'zh'],
+  german: ['da', 'de', 'en', 'es', 'fi', 'el', 'he', 'hu', 'it', 'nl', 'no', 'pt', 'sv'],
+  french: ['fr', 'tl', 'pt-br'],
+  russian: ['hr', 'ru', 'lt'],
+  czech: ['cs', 'sk'],
+  polish: ['pl'],
+  icelandic: ['is']
+};
+
+function langToTypeMap(mapping) {
+  var ret = {};
+  forEach(mapping, function (langs, type) {
+    forEach(langs, function (lang) {
+      ret[lang] = type;
+    });
+  });
+  return ret;
+}
+
+// Trim a string.
+var trimRe = /^\s+|\s+$/g;
+function trim(str) {
+  return replace.call(str, trimRe, '');
+}
+
+function pluralTypeName(locale) {
+  var langToPluralType = langToTypeMap(pluralTypeToLanguages);
+  return langToPluralType[locale]
+    || langToPluralType[locale.split(/-/, 1)[0]]
+    || langToPluralType.en;
+}
+
+function pluralTypeIndex(locale, count) {
+  return pluralTypes[pluralTypeName(locale)](count);
+}
+
+var dollarRegex = /\$/g;
+var dollarBillsYall = '$$';
+var tokenRegex = /%\{(.*?)\}/g;
+
+// ### transformPhrase(phrase, opts, locale)
+//
+// Takes a phrase string and transforms it by choosing the correct
+// plural form and interpolating it.
+//
+//     transformPhrase('Hello, %{name}!', {name: 'Spike'});
+//     // "Hello, Spike!"
+//
+// The correct plural form is selected if opts.smart_count
+// is set. You can pass in a number instead of an `opts` Object as a shortcut
+// for `smart_count`.
+//
+//     transformPhrase('%{smart_count} new messages |||| 1 new message', {smart_count: 1}, 'en');
+//     // "1 new message"
+//
+//     transformPhrase('%{smart_count} new messages |||| 1 new message', {smart_count: 2}, 'en');
+//     // "2 new messages"
+//
+//     transformPhrase('%{smart_count} new messages |||| 1 new message', 5, 'en');
+//     // "5 new messages"
+//
+// You should pass in a third argument, the locale, to specify the correct plural type.
+// It defaults to `'en'` with 2 plural forms.
+function transformPhrase(phrase, opts, locale) {
+  if (typeof opts === 'undefined' || opts === null) {
+    return phrase;
+  }
+
+  var result = phrase;
+
+  // allow number as a pluralization shortcut
+  var options = typeof opts === 'number' ? { smart_count: opts } : opts;
+
+  // Select plural form: based on a phrase text that contains `n`
+  // plural forms separated by `delimeter`, a `locale`, and a `opts.smart_count`,
+  // choose the correct plural form. This is only done if `count` is set.
+  if (options.smart_count != null && result) {
+    var texts = result.split(delimeter);
```

Comment:
While that might improve performance, it might also collect lots of memory. I would not expect a translation utility library to do caching for me.

How about we allow arrays of strings instead of `||||`-delimited strings, too, so people can cache themselves?